src/targets/rtr.rs

```diff
@@ -35,13 +35,22 @@ use crate::manager::Component;
 pub struct Tcp {
     listen: Vec<SocketAddr>,
     unit: Link,
+
+    #[serde(default = "Tcp::default_history_size")]
```

Comment:
Note: to avoid the `Tcp` and `Tls` structs becoming out of sync, isn't the `Tls` one a superset of the `Tcp` one that could include it and use Serde flattening to preserve the flat list of kv pairs in the TOML file?

Reply:
I was thinking of making a separate struct with the shared config, but re-using`Tcp` is even better.